src/testFixtures/java/org/opensearch/knn/TestUtils.java

```diff
@@ -181,6 +188,43 @@ public static double calculateRecallValue(List<List<String>> searchResults, List
         return sum / recalls.size();
     }
 
+    public static Map<SpaceType, BiFunction<float[], float[], Float>> knnScoringBySpaceType = ImmutableMap.of(
+        SpaceType.L1,
+        KNNScoringUtil::l1Norm,
+        SpaceType.L2,
+        KNNScoringUtil::l2Squared,
+        SpaceType.LINF,
+        KNNScoringUtil::lInfNorm
+    );
+
+    public static PriorityQueue<DistVector> computeGroundTruthValues(
+        float[] queryVector,
+        int numDocs,
+        int k,
+        int dimension,
+        SpaceType spaceType
+    ) {
+
+        PriorityQueue<DistVector> pq = new PriorityQueue<>(k, new DistComparator());
+
+        for (int i = 0; i < numDocs; i++) {
+            float dist = 0.0f;
+            float[] indexVector = new float[dimension];
+            Arrays.fill(indexVector, (float) i);
+            if (spaceType != null) {
+                dist = knnScoringBySpaceType.get(spaceType).apply(queryVector, indexVector);
```

Comment:
I think we need a default case when spaceType doesn't have corresponding scoring function. I think getOrDefault plus simple lambda that returns constant should work